 * Serving Flask app "label_studio_ml.api" (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:9090/ (Press CTRL+C to quit)
127.0.0.1 - - [17/Jul/2025 19:43:22] "GET /health HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:43:22] "POST /setup HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:43:49] "GET /health HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:43:49] "POST /setup HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:43:49] "GET /health HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:43:49] "POST /setup HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:44:05] "POST /webhook HTTP/1.1" 201 -
127.0.0.1 - - [17/Jul/2025 19:44:05] "POST /webhook HTTP/1.1" 201 -
Process Process-2:
Process Process-1:
Traceback (most recent call last):
Traceback (most recent call last):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 315, in _bootstrap
    self.run()
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 76, in job
    with self.start_run(event, data, job_id):
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 76, in job
    with self.start_run(event, data, job_id):
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/contextlib.py", line 117, in __enter__
    return next(self.gen)
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 168, in start_run
    job_dir = self._job_dir(job_id)
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 168, in start_run
    job_dir = self._job_dir(job_id)
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 178, in _job_dir
    return os.path.join(self.model_dir, str(job_id))
  File "/Users/c_fur1/Library/Python/3.9/lib/python/site-packages/label_studio_ml/model.py", line 178, in _job_dir
    return os.path.join(self.model_dir, str(job_id))
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/posixpath.py", line 76, in join
    a = os.fspath(a)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
TypeError: expected str, bytes or os.PathLike object, not NoneType
127.0.0.1 - - [17/Jul/2025 19:44:07] "GET /health HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:44:07] "POST /setup HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:46:10] "GET /health HTTP/1.1" 200 -
127.0.0.1 - - [17/Jul/2025 19:46:10] "POST /setup HTTP/1.1" 200 -
